wikimedia/wikipedia-ios-components
# Watchlist API Calls

**Phabricator:** 
https://phabricator.wikimedia.org/T334975

### Notes
This PR adds an initial `WKWatchlistViewModel` and `WKWatchlistViewController`. It has some ugly UI to demonstrate how to make the calls to the `WKData` layer. It also leans on the `WKData` mocked network service to display fake watchlist data, and to fake out POST calls.

** Test Steps **
1. Run in the Demo app (light theme), tap the Watchlist button.
2. You will see some faked watchlist data in a list. Tapping the Action button will present an action sheet of things you can try posting (watch, unwatch, change expiry, undo, rollback, etc.). Note that tapping the Action button fetches the watched status, which is randomly returned as true or false in the mocked network service. This is why the action sheet options randomly change. 

**Note**: We should first merge https://github.com/wikimedia/wikipedia-ios-data/pull/2, then change the data dependency in Package.swift to the `main` branch, before merging this one.
